src/games/strategy/triplea/ui/MapPanel.java

```diff
@@ -148,6 +161,30 @@ public void mouseReleased(final MouseEvent e) {
           notifyUnitSelected(tuple.getSecond(), tuple.getFirst(), md);
         }
       }
+
+      private boolean is4Pressed = false;
+      private boolean is5Pressed = false;
+      private int lastActive = -1;
+
+      @Override
+      public void mousePressed(final MouseEvent e) {
+        is4Pressed = e.getButton() == 4 ? true : is4Pressed;
```

Comment:
Okay, it's not clear why though. A comment in the code to explain would be useful.